tests/31sync/09archived.pl

```diff
@@ -56,16 +55,67 @@
       })->then( sub {
          my ( $body ) = @_;
 
+         log_if_fail "sync result", $body;
+
          my $room = $body->{rooms}{leave}{$room_id};
          assert_json_keys( $room, qw( timeline state ));
 
          @{ $room->{state}{events} } == 0
             or die "Expected no state events";
 
+         # we should see our own leave event
+         assert_eq(scalar @{ $room->{timeline}{events} }, 1,
+                   "timeline events");
+         my $ev = $room->{timeline}{events}[0];
+         assert_eq($ev->{type}, "m.room.member", "event type");
+         assert_eq($ev->{content}{membership}, "leave", "membership");
+
          Future->done(1);
       });
    };
 
+test "We should see our own leave event, even if history_visibility is " .
+    "restricted (SYN-662)",
```

Comment:
BTW, don't feel you have to stick to a strict 80-column limit just for strings like this; they're fine to spill over as there's nothing else semantically "important" out there.